client/src/modules/cohort/index.jsx

```diff
@@ -76,24 +81,45 @@ class Cohort extends PureComponent {
       cohortDetails: cohort,
       match: {
         params: { id }
-      }
+      },
+      members
     } = this.props;
     const {
       cohortDetails: previousCohort,
       currentUser: { id: userId },
       match: {
         params: { id: previousId }
-      }
+      },
+      members: previousMembers
     } = previousProps;
+    const { isDisabled } = this.state;
 
     if (id !== previousId) {
-      leaveRoom(Routes.COHORT, previousId, userId);
+      leaveRoom(Routes.COHORT, previousId, userId)(isDisabled);
       this.fetchData();
     }
 
     if (previousCohort && cohort && previousCohort.name !== cohort.name) {
       this.handleBreadcrumbs();
     }
+
+    if (previousCohort && cohort) {
```

Comment:
more descriptive state

Reply:
I made some more descriptive state in other conditions but here I can't think of something sensible: `cohortWasAndIs`, `cohortPersistButIstPropsCouldChange` the sound very stupid

Reply:
`cohortUpdated` and check if one `if` can be removed

Reply:
maybe "cohortDataStillAccessible" ?